servicetalk-client-api/src/test/java/io/servicetalk/client/api/LimitingConnectionFactoryFilterTest.java

```diff
@@ -76,17 +70,18 @@ public void setUp() {
     }
 
     @Test
-    public void enforceMaxConnections() throws Exception {
+    void enforceMaxConnections() throws Exception {
         ConnectionFactory<String, ? extends ListenableAsyncCloseable> cf =
                 makeCF(LimitingConnectionFactoryFilter.withMax(1), original);
-        cf.newConnection("c1", null).toFuture().get();
-        expectedException.expect(ExecutionException.class);
-        expectedException.expectCause(instanceOf(ConnectException.class));
-        cf.newConnection("c2", null).toFuture().get();
+        Exception e = assertThrows(ExecutionException.class, () -> {
+            cf.newConnection("c1", null).toFuture().get();
+            cf.newConnection("c2", null).toFuture().get();
+        });
```

Comment:
The first `newConnection` invocation is not expected to throw:
```suggestion
        cf.newConnection("c1", null).toFuture().get();
        Exception e = assertThrows(ExecutionException.class, () -> cf.newConnection("c2", null).toFuture().get());
```